source/core/slang-uint-set.h

```diff
@@ -101,6 +90,9 @@ class UIntSet
         /// Add a value
     inline void add(UInt val);
     inline void add(const UIntSet& val);
+    inline void add(const List<Element>& other);
```

Comment:
This reveals the internal implementation, again.
Can you avoid having this?

Reply:
This function does not reveal the internal implementation:
```
inline void UIntSet::add(const List<Element>& other)
{
    for (auto i : other)
        add((UInt)i);
}
```

it just does `add` on a list for convience.

Reply:
This function reveals that the class is internally using `List<Element>`. It creates an unnecessary dependency between the calling site and `List` and `Element`.
If we want to refactor the class later, as an example, we wouldn't be able to change Element nor List because calling sites will use those types.
This is bad from a class design perspective.

Reply:
Element is wrong, you are correct, this is a type-o.

Reply:
This utility function is not a good thing to have in general.
If we were able to use STL, there is for_each to do this type of things.

I am just letting you know that it is better not to have these things.
But I wouldn't object too hard for the connivence when there isn't for_each.

Reply:
I agree that this function is likely a bad idea from an STL design (not general enough)

maybe change this function to something like:
```
template<typename T>
inline void UIntSet::add(const T& other)
{
    for (auto i : other)
        add((UInt)i);
}
```

Reply:
I am OK with having dedicated `add` that allows us to use a more specialized implementation instead of relying on iterators too heavily.

If we are to follow existing container classes, the `add` here should be renamed to `addRange()` to signify that the operand is a collection.